File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        HashMap<Character, Character> hashmap = new HashMap<>();
        Stack<Character> stack = new Stack<>();

        hashmap.put('}','{');
        hashmap.put(')','(');
        hashmap.put(']','[');

        for(int i = 0; i < s.length(); i++) {
            if(hashmap.containsKey(s.charAt(i))) {
                if(stack.isEmpty() || !stack.pop().equals(hashmap.get(s.charAt(i)))) {
                    return false;
                }
            }
            else {
                stack.push(s.charAt(i));
            }
        }

        if(stack.isEmpty()) {
            return true;
        }
        return false;

    }
}

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        parent = {'}':'{', ']':'[', ')':'('}
        stack = []

        for i in s:
            if i in parent:
                if not stack or stack.pop() != parent[i]:
                    return False
            else:
                stack.append(i)
        if stack:
            return False
        return True